src/test/addrman_tests.cpp

```diff
@@ -34,7 +34,7 @@ class CAddrManTest : public CAddrMan
     //! Ensure that bucket placement is always the same for testing purposes.
     void MakeDeterministic()
     {
-        nKey.SetNull();
+        nKey = uint256{1};
```

Comment:
The same `MakeDeterministic()` also exists in `net_tests.cpp` and should be changed there too to prevent `net_tests/caddrdb_read` from failing (though I'm not sure why these tests are there, looks to me like they should be integrated into `addrman_tests.cpp` or removed if redundant).

Reply:
Oh good spot! There's also a deterministic addrman in the fuzzers.

I've added a `deterministic` argument to the `CAddrMan` ctor (similar to `TxRequestTracker`) to consolidate these into one place, so that the tests can just ask for a deterministic addrman rather than constructing and addrman and then updating its internals.